### DISCUSSION TITLE
how can I understand the meanning of cartesian_coeff ,when set cartesian_coeff(5)  to zero ,what does that mean

### DISCUSSION THREAD

**Discussion by dbdxnuliba** [tesseract-robotics/tesseract_ros#188] [Q&A]
in puzzle_piece_auxillary_axes_example
  // Create TrajOpt Profile
  auto trajopt_plan_profile = std::make_shared<tesseract_planning::TrajOptDefaultPlanProfile>();
  trajopt_plan_profile->cartesian_coeff = Eigen::VectorXd::Constant(6, 1, 5);
  trajopt_plan_profile->cartesian_coeff(3) = 2;
  trajopt_plan_profile->cartesian_coeff(4) = 2;
  trajopt_plan_profile->cartesian_coeff(5) = 0;
how can I understand the meanning of cartesian_coeff ,when set cartesian_coeff(5)  to zero ,what does that mean

**Comment by marip8** [comment]
`cartesian_coeff` are the weights for gradient descent with which the optimization will try to satisfy Cartesian position constraints for each dimension of the space (x, y, z, rx, ry, rz). The larger the weights, the more focus will be put on reducing errors for the quantities associated with those weights. Usually all are set to the same value, so the optimization tries to satisfy all elements of the space with equal strength. When a value is set to zero, it effectively allows the optimization to ignore errors in that dimension. For example, if the weights are [10, 10, 10, 10, 10, 0], then the optimization will unconstrain the rotation about rz. Similarly a weight vector of [0, 0, 0, 10, 10, 10] will allow the position to float freely while meeting orientation constraints.

**Reply by dbdxnuliba** [reply]
Thank you for your detailed and prompt reply!
